ktor-client/ktor-client-core/js/src/io/ktor/client/engine/js/JsPlatformApi.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.engine.js
+
+import io.ktor.client.engine.js.compatibility.*
+import io.ktor.util.*
+import io.ktor.utils.io.*
+import kotlinx.coroutines.*
+import org.w3c.dom.*
+import org.w3c.fetch.*
+import kotlin.js.*
+
+/**
+ * Platform specific API for JS
+ */
+public interface JsPlatformApi {
+
+    public fun fetch(url: String): Promise<Response>
```

Comment:
Do you need this function? Currently, only the second one with the `init: RequestInit` is called

Reply:
Since this is a public API, it's good to provide an API to fetch with the default request init args. Unfortunately, we can't use `dynamic` value in interfaces. That's why there are two methods.